commands/deploy_command.go

```diff
@@ -59,6 +59,14 @@ func (variable *listFlag) Set(value string) error {
 var modulesList listFlag
 var resourcesList listFlag
 
+var deployStrategies = map[string]func(*util.ProcessBuilder, map[string]interface{}) {
+	"blue-green": func(processBuilder *util.ProcessBuilder, optValues map[string]interface{}) {
```

Comment:
I do not like this approach. As I told before, this makes a mixture of commands (bg deploy + deploy + in future something more).
As this is a new approach for executing a command, let's do something like this. Intorduce a CommandStrategy class which is related to a specific command. This strategy should be able to define how the actions can be executed and how to define the getters.
What benefits gives you this approach:
- extensibility
- readability
- separation of concerns between different commands

Reply:
Also, each CommandStrategy should be able to extend the Help command.
About the command flags parsing, each additional strategy should bring some correction about the logic which parses and validates the command line options.
Finally, the command itself, should choose between each CommandStrategy(using some generic factory) which is to execute.